hack/update-bazel.sh

```diff
@@ -35,18 +39,132 @@ go install ./vendor/github.com/bazelbuild/bazel-gazelle/cmd/gazelle
+  local rule="k8s_${name}" # k8s_deepcopy, which copies out the file for a particular package
+
+  # look for packages that contain match
+  echo "Looking for packages with a $match comment in go code..."
+
+  white=($(find . -name *.go | \
+      grep -v "$pkg" | \
+      (xargs grep -l "$match" || true)))
+  want=()
+  for w in "${white[@]}"; do
+    if grep "$match" "$w" | grep -q -v "${match}false"; then
+      want+=( "$(dirname "$w" | $SED -e 's|./staging/src/|./vendor/|;s|^./||')" )
+    else
+      echo SKIP: $w, has only ${match}false tags
+    fi
+  done
+  want=($(IFS=$'\n ' ; echo "${want[*]}" | sort -u))
+  deps=()
+  for w in "${want[@]}"; do
+    deps+=( "./$w" )
+  done
+  deps=( $(go list -f "{{.ImportPath}}{{\"\n\"}}{{range .Deps}}{{.}}{{\"\n\"}}{{end}}" "${deps[@]}" ${extra:+"${extra[@]}"} | sort -u | grep -E '^k8s.io/kubernetes/' | $SED -e 's|^k8s.io/kubernetes/||') )
+
+  # Ensure that k8s_deepcopy_all() rule exists
+  dozer "new_load //build:deepcopy.bzl $all_rule" //build:__pkg__
+  ensure-rule //build "$all_rule" "${name}-sources"
+  dozer "set packages ${want[*]}" "//build:$name-sources"
+  dozer "set deps ${deps[*]}" "//build:$name-sources"
+  dozer "add srcs :$name-sources" //build:all-generated-sources
+  have=$(find . -name BUILD -or -name BUILD.bazel | (xargs grep -l "$rule(" || true))
+  if [[ -n "$have" ]]; then
+    echo Deleting existing "$rule" commands...
+    $SED -i -e "/^$rule/d" $have
+  fi
+
+  echo "Adding $rule() rules"
+  for w in "${want[@]}"; do
+    if [[ $w == */$pkg ]]; then
+      echo "ERROR: $pkg should not generate itself"
+      exit 1
+    fi
+    if [[ -f $w/BUILD.bazel ]]; then
+      echo "$rule(outs=[\"${out}\"])" >> $w/BUILD
+    elif  [[ -f $w/BUILD ]]; then
+      echo "$rule(outs=[\"${out}\"])" >> $w/BUILD
+    else
+      echo cannot find build file for $w
+      continue
+    fi
+    dozer "new_load //build:deepcopy.bzl $rule" //$w:__pkg__
+  done
+  echo "Deleting $out files"
+  find . -iname "${out}" | xargs rm
+}
+update-k8s-gengo deepcopy k8s.io/code-generator/cmd/deepcopy-gen zz_generated.deepcopy.go '+k8s:deepcopy-gen='
+extra_conv=($( \
+  find . -name *.go | \
+  xargs grep "+k8s:conversion-gen=" | \
+  grep -v conversion-gen=false | grep -v cmd/conversion-gen | \
+  sed -e 's|^[^=]*=||' | sort -u | $SED -e 's|k8s.io/kubernetes/|./|;s|\(^[^.]\)|./vendor/\1|'))
+update-k8s-gengo conversion k8s.io/code-generator/cmd/conversion-gen zz_generated.conversion.go '+k8s:conversion-gen=' "${extra_conv[@]}"
+update-k8s-gengo defaulter k8s.io/code-generator/cmd/defaulter-gen zz_generated.defaults.go '+k8s:defaulter-gen='
+echo Running gazelle to cleanup any changes...
+gazelle-fix
+
+if ! which -s bazel; then
+  return 0  # We should have already generated these files
+fi
+
+find . -name BUILD -or -name BUILD.bazel | xargs sed -i '' -e 's|@io_bazel_rules_go//go:def.bzl|//build:go.bzl|g'
+
+# Add dependencies from generated files
```

Comment:
oh wow, the kubernetes codegen is gross and rather bazel-hostile. :\

Reply:
Autogo might be able to help here